### PR TITLE
[FEAT] Adding HUD in goblin retreat

### PR DESCRIPTION
**Pull request by HBandesh**
# Description

In this PR, We have added the HUD in the goblin retreat. Some actions are not available in the goblin retreat HUD like saving, settings, buildings button, viewing/claiming skills/achievements, and placing items on the map from the chest.

image

Fixes #issue

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/bumpkins/components/AchievementBadges.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ export const AchievementBadges: React.FC<{
   });
 
   if (badges.length === 0) {
-    return null;
+    return <div className="text-xs">No achievments claimed.</div>;
```
Comment (mGranTY) [inline src/features/bumpkins/components/AchievementBadges.tsx:28]: typo here

"No achievments"

**Review by Rofrtd — APPROVED** [review]
Awesome work on this one! 

I've tested this locally and everything is working well.. 

🚀

**Review by spencerdezartsmith — COMMENTED** [review]
Mate... this looks like it was a doozy! I have left some comments regarding a couple of things. 

TBH I am a little concerned at how complicated this makes some of the components. There are a lot of different conditions that now exist inside of these components like `Inventory` and its children.

I am a little unsure about why we need to pass down an entire context and then have all these union types when all the Retreat is is basically a "read only" version of the Bumpkin and the Inventory. 

Could we possibly just make these components dumber and pass state and callbacks as props? 

Maybe I am missing something? 🥴 😅

**Review by adamhannigan — CHANGES_REQUESTED** [review]
Time to make some Goblin actions!

These components are getting complex at the moment, juggling multiple game contexts. 

We need to do the following to ensure this stays maintainable:

1. Go for an MVP (we don't need to click into a Bumpkin profile, don't need shortcuts in Goblin village, settings etc.)
2. Take a readonly/controlled component approach to Balance + Inventory components. Trying to handle 2 different states (goblin or game) from within these components is mixing up our UI and architecture logic.

@Rofrtd @HBandesh  Let's dive into this once you pop in.

**Comment by adamhannigan** [comment]
https://github.com/sunflower-land/sunflower-land/pull/1855

**Pull request closed without merging**